test/exercise/arrays/solution.rb

```diff
@@ -1,12 +1,34 @@
 module Exercise
   module Arrays
     class << self
+      def find_max(array)
+        max_value = array.first
+        array.each do |value|
+          max_value = value if value > max_value
+        end
+
+        max_value
+      end
+
       def replace(array)
-        array
+        max_value = find_max(array)
+        array.map { |value| value.positive? ? max_value : value }
       end
 
-      def search(_array, _query)
-        0
+      def search(array, search_element, index = 0)
+        middle_index = (array.size / 2).round
```

Comment:
Move this line after a block of 3 `return`s. We don't need to count `middle_index` if one of those `return`s got triggered.